### PR TITLE
backport: Merge #18670,18806,19060,19299,9828,20064,20105,18808

### PR DESCRIPTION
**Pull request by vijaydasmp**
…lear()

69ffddc83e0f3e265bf6cf7ae31489ae629fe6be refactor: Remove unused methods CBloomFilter::reset()/clear() (Sebastian Falbesoner)

Pull request description:

  The method `CBloomFilter::reset()` was introduced by commit d2d7ee0e863b286e1c9f9c54659d494fb0a7712d in 2015, but was never ever used, as far as I could find. As discovered by MarcoFalke, the method `clear()` is also unused outside of unit tests and is hence also removed.

ACKs for top commit:
  MarcoFalke:
    re-ACK 69ffddc83e0f3e265bf6cf7ae31489ae629fe6be
  jonatack:
    ACK 69ffddc83e0f3e2, code review, compiled a fuzz build and started the bloom_filter fuzz test as a sanity check.
  promag:
    ACK 69ffddc83e0f3e265bf6cf7ae31489ae629fe6be.

Tree-SHA512: 6c53678545ad8e2fa1ffc0a8838e450462f26748a60632f738dc020f0eb494ae2c32841e6256e266ed9140177257a78b707123421942f3819a14ffcb9a99322f

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: test/functional/test_framework/mininode.py [hunk 1]
```diff
@@ -437,7 +437,7 @@ def on_version(self, message):
 
     # Connection helper methods
 
-    def wait_until(self, test_function, timeout):
+    def wait_until(self, test_function, timeout=60):
```
Comment (UdjinM6) [inline test/functional/test_framework/mininode.py:440]: 19060: needs 18808

File: doc/REST-interface.md [hunk 2]
```diff
@@ -72,7 +72,6 @@ Only supports JSON as output format.
 * pruned : (boolean) if the blocks are subject to pruning
 * pruneheight : (numeric) highest block available
 * softforks : (array) status of softforks in progress
-* bip9_softforks : (object) status of BIP9 softforks in progress
```
Comment (UdjinM6) [inline doc/REST-interface.md:75]: 19390: too early for this, we still have `bip9_softforks`

File: src/test/fuzz/locale.cpp [hunk 3]
```diff
@@ -52,7 +52,6 @@ FUZZ_TARGET(locale)
     const bool parseint64_without_locale = ParseInt64(random_string, &parseint64_out_without_locale);
     const int64_t atoi64_without_locale = atoi64(random_string);
     const int atoi_without_locale = atoi(random_string);
-    const int64_t atoi64c_without_locale = atoi64(random_string.c_str());
```
Comment (UdjinM6) [inline src/test/fuzz/locale.cpp:55]: 19750: drop it here, it is a part of #5110

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/REST-interface.md (hunk 2 above)
Comment (vijaydasmp) [reply doc/REST-interface.md]: dropped 19390

**Comment by UdjinM6** [comment]
pls see 22a28658c919343b0a46382c5097fd59a5d08e3e and aac25888ff09400bf76ada563d1c364d7d9e90c5 (make sure to move "merge 19060" commit below "merge 18808" one before squashing fixes!)

**Comment by UdjinM6** [comment]
is there a reason to add 18839 here? it's a part of (an already approved) #5122, this will cause conflicts

**Comment by vijaydasmp** [comment]
> "merge 19060" commit below "merge 18808"



> is there a reason to add 18839 here? it's a part of (an already approved) #5122, this will cause conflicts

my bad, removed that

**Comment by vijaydasmp** [comment]
> pls see [22a2865](https://github.com/dashpay/dash/commit/22a28658c919343b0a46382c5097fd59a5d08e3e) and [aac2588](https://github.com/dashpay/dash/commit/aac25888ff09400bf76ada563d1c364d7d9e90c5) (make sure to move "merge 19060" commit below "merge 18808" one before squashing fixes!)

done

**Comment by UdjinM6** [comment]
> > pls see [22a2865](https://github.com/dashpay/dash/commit/22a28658c919343b0a46382c5097fd59a5d08e3e) and [aac2588](https://github.com/dashpay/dash/commit/aac25888ff09400bf76ada563d1c364d7d9e90c5) (make sure to move "merge 19060" commit below "merge 18808" one before squashing fixes!)
> 
> done

19060 looks good but no changes from 22a2865 were applied to 18808 it seems

**Comment by vijaydasmp** [comment]
> > > pls see [22a2865](https://github.com/dashpay/dash/commit/22a28658c919343b0a46382c5097fd59a5d08e3e) and [aac2588](https://github.com/dashpay/dash/commit/aac25888ff09400bf76ada563d1c364d7d9e90c5) (make sure to move "merge 19060" commit below "merge 18808" one before squashing fixes!)
> >
> > 
> > done
> 
> 19060 looks good but no changes from [22a2865](https://github.com/dashpay/dash/commit/22a28658c919343b0a46382c5097fd59a5d08e3e) were applied to 18808 it seems

Hello @UdjinM6 changes from [22a2865] done now

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , please review

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**